diff -ur bcti-sdk-linux-5.26.2.orig/src/Daemon/Daemon.h bcti-sdk-linux-5.26.2.new/src/Daemon/Daemon.h
--- bcti-sdk-linux-5.26.2.orig/src/Daemon/Daemon.h	2018-07-18 12:39:37.000000000 -0700
+++ bcti-sdk-linux-5.26.2.new/src/Daemon/Daemon.h	2019-04-10 09:46:59.945242810 -0700
@@ -278,6 +278,7 @@
 public:
 	static cJSON* GetCatConfObject(const CatConf* cc, const char* urlString = 0);
 	static cJSON* GetCatConfObject(const CatConf* cc, std::string str);
+	static cJSON* GetCatConfArray(const CatConf* cc);
 
 	static cJSON* GetWhoIsInfoObject(const WhoisBasicData& data);
 
diff -ur bcti-sdk-linux-5.26.2.orig/src/Daemon/DaemonUrl.cpp bcti-sdk-linux-5.26.2.new/src/Daemon/DaemonUrl.cpp
--- bcti-sdk-linux-5.26.2.orig/src/Daemon/DaemonUrl.cpp	2018-04-05 12:28:02.000000000 -0700
+++ bcti-sdk-linux-5.26.2.new/src/Daemon/DaemonUrl.cpp	2019-04-10 09:47:09.617076070 -0700
@@ -65,6 +65,22 @@
 	return GetCatConfObject(cc, str.empty() ? 0 : str.c_str());
 }
 
+cJSON* DaemonUrl::GetCatConfArray(const CatConf* cc)
+{
+	cJSON* catidsArray = cJSON_CreateArray();
+
+	int i = 0;
+	for (; i < UrlInfo::MAX_CATCONF; ++i)
+	{
+		if ( i > 0 && ((0 == cc[i].category) || (0 == cc[i].confidence)))
+			break;
+		Json::AddNumber(catidsArray, cc[i].category);
+	}
+
+	return catidsArray;
+}
+
+
 cJSON* DaemonUrl::GetWhoIsInfoObject(const WhoisBasicData& data)
 {
 	cJSON* whoIsObj = cJSON_CreateObject();
@@ -120,7 +136,8 @@
 			"urls" : ["url", "..."],
 			"?cached" : false,
 			"?uncat_info" : false,
-			"?timer" : false
+			"?timer" : false,
+			"?catids" : false
 		)
 	)
 
@@ -128,6 +145,7 @@
 	GET_JSON_BOOL(json, "cached", isCacheLookup, false)
 	GET_JSON_BOOL(json, "uncat_info", addUncatInfo, false)
 	GET_JSON_BOOL(json, "timer", timeOperation, false)
+	GET_JSON_BOOL(json, "catids", categoryIds, false)
 
 	UrlInfoRequest reqs[urlCount];
 	for (int i = 0; i < urlCount; ++i)
@@ -159,9 +177,16 @@
 			cJSON* infoObj = cJSON_CreateObject();
 
 			Json::AddNumber(infoObj, "reputation", req.data.reputation);
-			Json::AddObject(
-				infoObj, "categories",
-				GetCatConfObject(req.data.cc, addUncatInfo ? req.url : 0));
+			if(categoryIds){
+				Json::AddArray(
+					infoObj, "catids",
+					GetCatConfArray(req.data.cc));
+
+			}else{
+				Json::AddObject(
+					infoObj, "categories",
+					GetCatConfObject(req.data.cc, addUncatInfo ? req.url : 0));
+			}
 
 			Json::AddBool(infoObj, "a1cat", req.data.flags & UrlInfo::ONE_CAT);
 
