--- bcti-sdk-linux-5.27.1.current/src/IsisUrl/URL.cpp	2019-04-03 14:00:29.000000000 -0700
+++ bcti-sdk-linux-5.27.1/src/IsisUrl/URL.cpp	2019-05-23 08:22:24.000000000 -0700
@@ -291,7 +291,7 @@
 		return false;
 
 	const char *s;
-//	const char *s1;
+	const char *s1;
 	const char *l_hostinfo;
 	char uri_char[URL_SIZE];
 	const char *uri = (const char *)uri_char;
@@ -421,8 +421,9 @@
 			else
 			{
 				// Only IPv6 addresses may have more than one colon.
-				if ((s = strchr(l_hostinfo, ':')) && strchr(s + 1, ':'))
+				if ((s = strchr(l_hostinfo, ':')) && (s1 = strchr(s + 1, ':')) && ((s1 - s) == 5))
 					return false;
+				s = 0;
 			}
 			SetHostname(l_hostinfo, s ? s - l_hostinfo : hostLen, &ipAddr);
 			if (!*m_hostname)
