diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkCdb/CdbConnectionTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkCdb/CdbConnectionTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkCdb/CdbConnectionTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkCdb/CdbConnectionTask.cpp	2019-06-14 10:23:46.000000000 -0400
@@ -44,7 +44,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	else
 	{
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkCdb/CdbStatsTask.h bcti-sdk-linux-5.27.2/src/IsisSdkCdb/CdbStatsTask.h
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkCdb/CdbStatsTask.h	2019-04-03 19:45:52.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkCdb/CdbStatsTask.h	2019-06-14 10:24:07.000000000 -0400
@@ -45,7 +45,12 @@
 		if (m_query.HasQuery())
 		{
 			IncrementNet();
-			return m_query.Start(this);
+			if(m_query.Start(this) == false){
+				OnError();
+				return false;
+			}else{
+				return true;
+			}
 		}
 		else
 		{
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkCore/StatsTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkCore/StatsTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkCore/StatsTask.cpp	2019-04-18 17:05:37.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkCore/StatsTask.cpp	2019-06-14 10:22:04.000000000 -0400
@@ -51,7 +51,12 @@
 	if (m_query.HasQuery())
 	{
 		m_query.SetProtocol(m_proto);
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkFile/Md5FileTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkFile/Md5FileTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkFile/Md5FileTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkFile/Md5FileTask.cpp	2019-06-14 10:22:23.000000000 -0400
@@ -122,7 +122,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpGeoTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpGeoTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpGeoTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpGeoTask.cpp	2019-06-14 10:20:07.000000000 -0400
@@ -103,7 +103,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpReputationTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpReputationTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpReputationTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpReputationTask.cpp	2019-06-14 10:19:53.000000000 -0400
@@ -101,7 +101,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpThreatListTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpThreatListTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpThreatListTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpThreatListTask.cpp	2019-06-14 10:20:26.000000000 -0400
@@ -45,7 +45,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	else
 	{
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpThreatTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpThreatTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpThreatTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpThreatTask.cpp	2019-06-14 10:20:44.000000000 -0400
@@ -111,7 +111,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpUpdateTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpUpdateTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkIp/IpUpdateTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkIp/IpUpdateTask.cpp	2019-06-14 10:20:57.000000000 -0400
@@ -54,7 +54,12 @@
 	Log(LOG_Trace, "IpUpdateTask: DoNetwork");
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	else
 	{
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkMobileApps/MobileTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkMobileApps/MobileTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkMobileApps/MobileTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkMobileApps/MobileTask.cpp	2019-06-14 10:19:02.000000000 -0400
@@ -128,7 +128,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkPhishing/PhishingRequestTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkPhishing/PhishingRequestTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkPhishing/PhishingRequestTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkPhishing/PhishingRequestTask.cpp	2019-06-14 10:21:42.000000000 -0400
@@ -91,7 +91,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	else
 	{
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkPhishing/PhishingResponseTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkPhishing/PhishingResponseTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkPhishing/PhishingResponseTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkPhishing/PhishingResponseTask.cpp	2019-06-14 10:21:18.000000000 -0400
@@ -69,7 +69,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	else
 	{
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkPhishing/PhishingScoreTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkPhishing/PhishingScoreTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkPhishing/PhishingScoreTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkPhishing/PhishingScoreTask.cpp	2019-06-14 10:21:30.000000000 -0400
@@ -146,7 +146,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/CatListTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkUrl/CatListTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/CatListTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/CatListTask.cpp	2019-06-14 10:22:38.000000000 -0400
@@ -45,7 +45,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/UrlInfoTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkUrl/UrlInfoTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/UrlInfoTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/UrlInfoTask.cpp	2019-06-14 10:23:01.000000000 -0400
@@ -173,7 +173,12 @@
 	if (m_query.HasQuery())
 	{
 		IncrementNet();
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/UrlRepTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkUrl/UrlRepTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/UrlRepTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/UrlRepTask.cpp	2019-06-14 10:23:12.000000000 -0400
@@ -77,7 +77,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/UrlUpdateTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkUrl/UrlUpdateTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/UrlUpdateTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/UrlUpdateTask.cpp	2019-06-14 10:22:50.000000000 -0400
@@ -50,7 +50,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	return false;
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/WhoisBasicTask.cpp bcti-sdk-linux-5.27.2/src/IsisSdkUrl/WhoisBasicTask.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/WhoisBasicTask.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/WhoisBasicTask.cpp	2019-06-14 10:23:24.000000000 -0400
@@ -48,7 +48,12 @@
 {
 	if (m_query.HasQuery())
 	{
-		return m_query.Start(this);
+		if(m_query.Start(this) == false){
+			OnError();
+			return false;
+		}else{
+			return true;
+		}
 	}
 	else
 	{
