diff -ur bcti-sdk-linux-5.27.2.orig/src/Daemon/Daemon.cpp bcti-sdk-linux-5.27.2/src/Daemon/Daemon.cpp
--- bcti-sdk-linux-5.27.2.orig/src/Daemon/Daemon.cpp	2019-05-04 17:36:27.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/Daemon/Daemon.cpp	2019-06-14 11:23:01.000000000 -0400
@@ -253,6 +253,10 @@
 	Json::StringFormattedToObject(jUrlDb, "version", "%d.%d.%d", major, minor, update);
 	Json::Number(jUrlDb, "type", (size_t)config->GetUrlDatabaseType());
 	Json::Number(jUrlDb, "update_frequency_mins", config->GetUrlUpdateTimeMins());
+	Json::Number(jUrlDb, "url_cache_maximum_size", config->GetUrlCacheSizeMB() * 1048576);
+	int cacheCount = 0;
+	GetUrlCacheRecordCount(cacheCount);
+	Json::Number(jUrlDb, "url_cache_current_size", cacheCount);
 	char url[PATH_MAX + 1] = { 0 };
 	GetUrlFileName(url, PATH_MAX);
 	Json::String(jUrlDb, "file_path", url);
diff -ur bcti-sdk-linux-5.27.2.orig/src/Daemon/DaemonUrl.cpp bcti-sdk-linux-5.27.2/src/Daemon/DaemonUrl.cpp
--- bcti-sdk-linux-5.27.2.orig/src/Daemon/DaemonUrl.cpp	2019-04-24 12:51:40.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/Daemon/DaemonUrl.cpp	2019-06-14 15:18:57.000000000 -0400
@@ -223,4 +223,18 @@
 			timeOperation, tag
 		);
 }
+
+DAEMON_SDK_DEFN(DaemonUrl::SetUrlCacheClear)
+{
+	CHECK_JSON("")
+
+	if (Isis::SetUrlCacheClear())
+	{
+		cJSON* jsonResp = Json::Object();
+		Json::Bool(jsonResp, "cleared", true);
+		return jsonResp;
+	}
+	else
+		return Json::ErrorObject("cache clear failure");
+}
 }
diff -ur bcti-sdk-linux-5.27.2.orig/src/Daemon/DaemonUrl.h bcti-sdk-linux-5.27.2/src/Daemon/DaemonUrl.h
--- bcti-sdk-linux-5.27.2.orig/src/Daemon/DaemonUrl.h	2019-04-24 12:48:30.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/Daemon/DaemonUrl.h	2019-06-14 15:18:52.000000000 -0400
@@ -37,6 +37,7 @@
 	DAEMON_SDK_DECL(GetUrlInfo);
 	DAEMON_SDK_DECL(GetUrlReputation);
 	DAEMON_SDK_DECL(GetUrlWhoIs);
+	DAEMON_SDK_DECL(SetUrlCacheClear);
 };
 }
 #endif	// #define DaemonUrl_h
diff -ur bcti-sdk-linux-5.27.2.orig/src/Daemon/main.cpp bcti-sdk-linux-5.27.2/src/Daemon/main.cpp
--- bcti-sdk-linux-5.27.2.orig/src/Daemon/main.cpp	2019-06-13 15:23:19.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/Daemon/main.cpp	2019-06-14 11:18:07.000000000 -0400
@@ -402,6 +402,7 @@
 	s_sdkFns["url/submitphishquery"] = DaemonPhishing::SubmitPhishingQuery;
 	s_sdkFns["url/getwhoisinfo"] = DaemonUrl::GetUrlWhoIs;
 	s_sdkFns["url/getcatlist"] = DaemonUrl::GetUrlCategories;
+	s_sdkFns["url/setcacheclear"] = DaemonUrl::SetUrlCacheClear;
 	s_sdkFns["counters"] = Daemon::GetStatsCounters;
 	s_sdkFns["status"] = Daemon::GetStatus;
 
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/IsisSdkUrl.cpp bcti-sdk-linux-5.27.2/src/IsisSdkUrl/IsisSdkUrl.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/IsisSdkUrl.cpp	2019-04-03 19:45:56.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/IsisSdkUrl.cpp	2019-06-14 15:13:44.000000000 -0400
@@ -304,5 +304,22 @@
 	UrlDbManager::Instance()->LoadExisting();
 	return true;
 }
+
+bool GetUrlCacheRecordCount(int& count)
+{
+	count = 0;
+	if (!Configuration::Instance()->UrlEnabled())
+		return false;
+	count = UrlCache::Instance()->GetCacheRecordCount();
+	return true;
+}
+
+bool SetUrlCacheClear()
+{
+	if (!Configuration::Instance()->UrlEnabled())
+		return false;
+	UrlCache::Instance()->Initialize();
+	return true;
+}
 } // namespace Isis
 
diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/IsisSdkUrl.h bcti-sdk-linux-5.27.2/src/IsisSdkUrl/IsisSdkUrl.h
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkUrl/IsisSdkUrl.h	2019-04-03 19:45:52.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkUrl/IsisSdkUrl.h	2019-06-14 15:23:11.000000000 -0400
@@ -351,6 +351,18 @@
 	Returns true and the version numbers if a database is loaded
 */
 ISISAPI bool GetUrlDatabaseVersion(int& major, int& minor, int& update);
-} // namespace Isis
 
+//! GetUrlCacheRecordCount
+/*!
+	Returns true and the current number of entries in the cache.
+*/
+ISISAPI bool GetUrlCacheRecordCount(int& count);
+
+//! SetUrlCacheClear
+/*!
+	Clear URL cache
+*/
+ISISAPI bool SetUrlCacheClear();
+
+} // namespace Isis
 #endif // IsisSdk_IsisSdkUrl_h
