diff -ur bcti-sdk-linux-5.27.2.orig/src/IsisSdkCore/TCPConnection.cpp bcti-sdk-linux-5.27.2/src/IsisSdkCore/TCPConnection.cpp
--- bcti-sdk-linux-5.27.2.orig/src/IsisSdkCore/TCPConnection.cpp	2019-07-25 15:16:06.000000000 -0400
+++ bcti-sdk-linux-5.27.2/src/IsisSdkCore/TCPConnection.cpp	2019-08-07 12:39:15.000000000 -0400
@@ -322,6 +322,8 @@
 	if (IpAddress::Protocol::Undefined == GetProtocol())
 	{
 		Log(LOG_Error, "Cannot determine Internet Protocol for host %s", host.c_str());
+		m_addr.SetName(host);
+		m_port = port;
 		return ConnectError();
 	}
 
